packages/aws-cdk-lib/aws-stepfunctions-tasks/lib/bedrock/create-model-customization-job.ts

```diff
@@ -0,0 +1,479 @@
+   */
+  readonly jobName: string;
+
+  /**
+   * Tags to attach to the job.
+   * The maximum number of tags is 200.
+   *
+   * @default - no tags
+   */
+  readonly jobTags?: ITag[];
+
+  /**
+   * The S3 bucket configuration where the output data is stored.
+   *
+   * @see https://docs.aws.amazon.com/bedrock/latest/APIReference/API_OutputDataConfig.html
+   */
+  readonly outputData: BucketConfiguration;
+
+  /**
+   * The IAM role that Amazon Bedrock can assume to perform tasks on your behalf.
+   *
+   * For example, during model training, Amazon Bedrock needs your permission to read input data from an S3 bucket,
+   * write model artifacts to an S3 bucket.
+   * To pass this role to Amazon Bedrock, the caller of this API must have the iam:PassRole permission.
+   *
+   * @default - use auto generated role
+   */
+  readonly role?: iam.IRole;
+
+  /**
+   * The S3 bucket configuration where the training data is stored.
+   *
+   * @see https://docs.aws.amazon.com/bedrock/latest/APIReference/API_TrainingDataConfig.html
+   */
+  readonly trainingData: BucketConfiguration;
+
+  /**
+   * The S3 bucket configuration where the validation data is stored.
+   *
+   * The maximum number is 10.
+   *
+   * @see https://docs.aws.amazon.com/bedrock/latest/APIReference/API_Validator.html
+   */
+  readonly validationData: BucketConfiguration[];
+
+  /**
+   * The VPC used to protect the training data and customization job.
+   *
+   * The ENI used in this job will be created in one of the subnets among all the subnets possessed by the VPC.
+   *
+   * The minimum number of subnets for the VPC is 1.
+   * The maximum number of subnets for the VPC is 16.
+   *
+   * @see https://docs.aws.amazon.com/bedrock/latest/userguide/vpc-model-customization.html
+   *
+   * @default - no VPC configuration
+   */
+  readonly vpc?: ec2.IVpc;
+}
+
+/**
+ * A Step Functions Task to create model customization in Bedrock.
+ */
+export class BedrockCreateModelCustomizationJob extends sfn.TaskStateBase {
+
+  private static readonly SUPPORTED_INTEGRATION_PATTERNS: sfn.IntegrationPattern[] = [
+    sfn.IntegrationPattern.REQUEST_RESPONSE,
+    sfn.IntegrationPattern.RUN_JOB,
+  ];
+
+  protected readonly taskMetrics?: sfn.TaskMetricsConfig;
+  protected readonly taskPolicies?: iam.PolicyStatement[];
+
+  private readonly integrationPattern: sfn.IntegrationPattern;
+  private _role: iam.IRole;
+  private subnets: ec2.ISubnet[];
+  private securityGroup?: ec2.ISecurityGroup;
+
+  constructor(scope: Construct, id: string, private readonly props: BedrockCreateModelCustomizationJobProps) {
+    super(scope, id, props);
+    this.integrationPattern = props.integrationPattern ?? sfn.IntegrationPattern.REQUEST_RESPONSE;
+    this.subnets = props.vpc?.selectSubnets().subnets ?? [];
+    this.securityGroup = props.vpc ? new ec2.SecurityGroup(this, 'SecurityGroup', {
```

Comment:
🤔 Now I see the problem of just using the `vpc` property.
We don't allow much control over security groups to the users.
I guess we need to switch back using the `IBedrockCreateModelCustomizationJobVpcConfig` interface (please also add validation on `securityGroups` and `subnets` array lengths).